### DISCUSSION TITLE
Event driven applications

### DISCUSSION THREAD

**Discussion by cloudcompute** [redpanda-data/connect#1634] [Q&A] [answered]
Can we build event-driven applications using Benthos? To process the incoming data, is it possible to process it using [Knative Serving](https://github.com/knative/serving)

**Comment by mihaitodor** [comment, marked as answer]
@cloudcompute Sure! Have you seen my KnativeCon talk from earlier this year? The recording is [here](https://www.youtube.com/watch?v=3OaRXwcRJJk&list=PLj6h78yzYM2Mv__LXlfzAAw5WMUTzIvS8) and you can find the demo setup [here](https://github.com/mihaitodor/knative-benthos).

**Comment by cloudcompute** [comment]
@mihaitodor Thanks for the response. I went through the recording, Benthos and Knative Serving would probably prove a perfect combination.

The main point of concern is that there are 3 independent pieces: Nats API, Knative Serving, and Benthos. How can we connect all the three where Benthos sits in between the other two i.e. is there any Benthos API that we can use to create the pipelines from within our application? Or, we have to do all this using Yaml configuration?

If only through the yaml configurations, it'd probably be difficult to keep track of a great number of pipelines if it is a large application.

**Reply by mihaitodor** [reply]
Yep, there is a REST API if you use Benthos in streams mode (see [here](https://www.benthos.dev/docs/guides/streams_mode/using_rest_api/)). Additionally, if the yaml config becomes too unwieldily, you might want to look into generating it by using CUE: https://www.benthos.dev/docs/configuration/using_cue/

**Reply by cloudcompute** [reply]
Good enough, Thanks.
